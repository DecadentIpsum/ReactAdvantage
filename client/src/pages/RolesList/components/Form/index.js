import React, { Component } from 'react';

import TabView, { TabPanel } from 'components/TabView';
import ModalPopup from 'components/ModalPopup';
import Input from 'components/Input';
import Button from 'components/Button';
import Checkbox from 'components/Checkbox';
import Tree, { TreeNode } from 'components/Tree';
import validateForm from 'utils/validateForm';

import './index.css';

export default class Form extends Component {
    onSubmit = event => {
        event.preventDefault();
        const isFormValid = this.validateForm();

        if (!isFormValid) {
            this.setState({
                activeTabIndex: 0,
            });

            return;
        }

        const { onSubmit } = this.props;
        const { form } = this.state;
        let formValues = Object.keys(form).reduce((values, key) => {
            values[key] = form[key].value;

            return values;
        }, {});

        formValues.permissions = this.state.treeValue;

        onSubmit(formValues)
    }

    onChange = (name, value) => {
        const state = {...this.state};
        state.form[name].value = value;

        this.setState(state);
    }

    onTextChange = ({ target: { name, value } }) => {
        this.onChange(name, value);
    }

    onHide = () => {
        const { onHide } = this.props;
        onHide && onHide();
    }

    onCheck = data => {
        this.setState({
            treeValue: data,
        });
    }

    onTabChange = ({ index }) => {
        this.setState({
            activeTabIndex: index,
        });
    }

    getPopupTitle() {
        return this.props.role.roleName
            ? `Edit role: ${this.props.role.roleName}`
            : 'Create role';
    }

    defaultCheckedKeys = ['audit-logs'];

    state = {
        activeTabIndex: 0,
        treeValue: this.defaultCheckedKeys,
        form: {
            name: {
                value: this.props.role.roleName || '',
                validators: ['isFilled'],
                error: '',
            },
            default: {
                value: false,
            },
        },
    };

    treeData = [{
        key: 'Documents',
        title: 'Pages',
        children: [{
            key: 'Administration',
            title: 'Administration',
            children: [{
                key: 'audit-logs',
                title: 'Audit logs',
            }, {
                key: 'email-templates',
                title: 'Email Templates'
            }, {
                key: 'hangfire-dashboard',
                title: 'Hangfire Dashboard'
            }],
        }, {
            key: 'Resources',
            title: 'Resources',
            children: [{
                key: 'full-access',
                title: 'Full access',
            }]
        }],
    }, {
        key: 'Pictures',
        title: 'Editions',
        children: [{
            key: 'edit-roles',
            title: 'Edit Roles'
        }],
    }];

    validateForm() {
        const { form: defaultForm } = this.state;
        let { isFormValid, form } = validateForm(defaultForm);

        this.setState({ form });

        return isFormValid;
    }

    renderRolesForm() {
        const { form } = this.state;

        return (
            <form className="edit-role" onSubmit={this.onSubmit}>
                <div className="edit-role-input-wrapper">
                    <label
                        className="edit-role-label"
                        htmlFor="edit-role-name"
                    >
                        Role Name <sup>*</sup>
                     </label>
                    <Input
                        id="edit-role-name"
                        value={form.name.value}
                        error={form.name.error}
                        name="name"
                        onChange={this.onTextChange}
                        autoFocus
                    />
                </div>
                <div className="edit-role-input-wrapper edit-role-input-wrapper-checkbox">
                    <Checkbox
                        inputId="edit-role-default"
                        label="Default"
                        name="default"
                        checked={Boolean(form.default.value)}
                        value={form.default.value}
                        onChange={({ checked }) => this.onChange('default', checked)}
                    />
                    <div className="edit-role-form-hint">Assign to new users as default</div>
                </div>
            </form>
        );
    }

    renderTreeForm() {
        return (
            <form className="edit-role" onSubmit={this.onSubmit}>
                <Tree
                    checkable
                    defaultExpandAll
                    selectable={false}
                    defaultCheckedKeys={this.defaultCheckedKeys}
                    onCheck={this.onCheck}
                >
                    {this.renderTreeNodeContent(this.treeData)}
                </Tree>
            </form>
        );
    }

    renderActions() {
        return (
            <div>
                <Button
                    label="Cancel"
                    secondary
                    onClick={this.props.onHide}
                    className="edit-role-cancel-button"
                ></Button>
                <Button
                    label="Save"
                    onClick={this.onSubmit}
                    className="edit-role-save-button"
                ></Button>
            </div>
        );
    }

    renderTreeNodeContent(data) {
        return data.map(item => {
            if (item.children) {
                return (
                    <TreeNode
                        key={item.key}
                        title={item.title}
                    >
                        {this.renderTreeNodeContent(item.children)}
                    </TreeNode>
                );
            }

            return (
                <TreeNode key={item.key} title={item.title} />
            );
        });
    }

    render() {
        return (
            <ModalPopup
                visible={this.props.visible}
                onHide={this.onHide}
                footer={this.renderActions()}
                header={this.getPopupTitle()}
                inPortal={false}
                modal
            >
                <TabView
                    activeIndex={this.state.activeTabIndex}
                    onTabChange={this.onTabChange}
                >
                    <TabPanel header="Role Properties">
                        {this.renderRolesForm()}
                    </TabPanel>
                    <TabPanel header="Permissions">
                        {this.renderTreeForm()}
                    </TabPanel>
                </TabView>
            </ModalPopup>
        );
    }
}
