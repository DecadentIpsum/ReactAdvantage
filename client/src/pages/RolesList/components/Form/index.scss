@import "global-styles/common.scss";

.edit-role {
    width: 400px;
    margin: 0 auto;

    @media(max-width: 640px) {
        width: 100%;
    }
}

.edit-role-form-hint {
    margin-top: 5px;
    color: $grayShade;
    font-size: 0.8rem;
}

.edit-role-input-wrapper {
    margin-bottom: 20px;
}

.edit-role-label {
    display: block;
    margin-bottom: 10px;
    color: $blackShade;

    sup {
        color: $error;
    }
}

.edit-role-input-wrapper {
    label {
        color: $blackShade;
    }
}