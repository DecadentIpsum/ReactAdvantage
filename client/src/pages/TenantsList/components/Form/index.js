import React, { Component } from 'react';
import moment from 'moment';

import ModalPopup from 'components/ModalPopup';
import Input from 'components/Input';
import Checkbox from 'components/Checkbox';
import Button from 'components/Button';
import Dropdown from 'components/Dropdown';
import Calendar from 'components/Calendar';
import validateForm from 'utils/validateForm';

import './index.css';

export default class Form extends Component {
    onSubmit = event => {
        event.preventDefault();
        const isFormValid = this.validateForm();

        if (!isFormValid) {
            this.setState({
                activeTabIndex: 0,
            });

            return;
        }

        const { onSubmit } = this.props;
        const { form } = this.state;
        let formValues = Object.keys(form).reduce((values, key) => {
            values[key] = form[key].value;

            return values;
        }, {});

        formValues.subscription = moment(formValues.subscription).format('X');

        onSubmit(formValues)
    }

    onChange = (name, value) => {
        const state = {...this.state};
        state.form[name].value = value;

        this.setState(state);
    }

    onTextChange = ({ target: { name, value } }) => {
        this.onChange(name, value);
    }

    onDropdownChange = ({ value }) => {
        const state = {...this.state};
        state.form.editions.value = value;

        this.setState(state);
    }

    getPopupTitle() {
        return this.props.tenant.tenantName
            ? `Edit tenant: ${this.props.tenant.tenantName}`
            : 'Create tenant';
    }

    editions = [{
        label: 'Standard',
        value: 'Standard',
    }, {
        label: 'Pro',
        value: 'Pro',
    }]

    state = {
        form: {
            name: {
                value: this.props.tenant.tenantName || '',
                validators: ['isFilled'],
                error: '',
            },
            editions: {
                value: this.editions[0].value,
            },
            subscription: {
                value: new Date(),
                validators: ['isFilled'],
                error: '',
            },
            active: {
                value: false,
            },
        }
    };

    validateForm() {
        const { form: defaultForm } = this.state;
        let { isFormValid, form } = validateForm(defaultForm);

        this.setState({ form });

        return isFormValid;
    }

    renderForm() {
        const { form } = this.state;

        return (
            <form className="edit-tenant" onSubmit={this.onSubmit}>
                <div className="edit-tenant-input-wrapper">
                    <label
                        className="edit-tenant-label"
                        htmlFor="edit-tenant-name"
                    >
                        Name <sup>*</sup>
                    </label>
                    <Input
                        id="edit-tenant-name"
                        value={form.name.value}
                        error={form.name.error}
                        onChange={this.onTextChange}
                        name="name"
                        autoFocus
                    />
                </div>
                <div className="edit-tenant-input-wrapper">
                    <label className="edit-tenant-label">
                        Edition
                    </label>
                    <Dropdown
                        options={this.editions}
                        value={form.editions.value}
                        onChange={this.onDropdownChange}
                    />
                </div>
                <div className="edit-tenant-input-wrapper">
                    <label className="edit-tenant-label">
                        Subscription end date <sup>*</sup>
                    </label>
                    <Calendar
                        value={form.subscription.value}
                        minDate={new Date()}
                        onChange={({ value }) => this.onChange('subscription', value)}
                    />
                </div>
                <div className="edit-tenant-input-wrapper">
                    <Checkbox
                        inputId="edit-tenant-active"
                        label="Active"
                        name="active"
                        checked={Boolean(form.active.value)}
                        value={form.active.value}
                        onChange={({ checked }) => this.onChange('active', checked)}
                    />
                </div>
            </form>
        );
    }

    renderActions() {
        return (
            <div>
                <Button
                    secondary
                    onClick={this.props.onHide}
                >
                    Cancel
                </Button>
                <Button
                    onClick={this.onSubmit}
                >
                    Save
                </Button>
            </div>
        );
    }

    render() {
        return (
            <ModalPopup
                visible={this.props.visible}
                onHide={this.props.onHide}
                header={this.getPopupTitle()}
                footer={this.renderActions()}
                inPortal={false}
                modal
            >
                {this.renderForm()}
            </ModalPopup>
        );
    }
}
