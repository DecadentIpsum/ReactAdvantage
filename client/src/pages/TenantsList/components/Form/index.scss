@import "global-styles/common.scss";

.edit-tenant {
    width: 400px;
    margin: 0 auto;

    @media (max-width: 640px) {
        width: 100%;
    }
}

.edit-tenant-input-wrapper {
    margin-bottom: 20px;

    label {
        color: $blackShade;
    }
}

.edit-tenant-label {
    display: block;
    margin-bottom: 10px;

    sup {
        color: $error;
    }
}
