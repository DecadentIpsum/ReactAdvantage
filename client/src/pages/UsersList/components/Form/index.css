.edit-user {
  width: 400px;
  margin: 0 auto; }
  @media (max-width: 640px) {
    .edit-user {
      width: 100%; } }

.edit-user-upload {
  position: relative;
  float: left;
  width: 80px;
  height: 80px;
  border: 1px solid #B3B3AB;
  margin-bottom: 20px;
  margin-top: 6px;
  cursor: pointer; }
  @media (max-width: 640px) {
    .edit-user-upload {
      float: none;
      margin: 0 auto 20px; } }
  .edit-user-upload::after {
    content: '';
    clear: both;
    display: table-cell; }
  .edit-user-upload label {
    position: absolute;
    width: 100%;
    height: 100%;
    cursor: pointer; }
  .edit-user-upload img {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 95%;
    max-width: 95%;
    border-radius: 50%;
    pointer-events: none; }
  .edit-user-upload .ui-button {
    display: none; }

.users-edit-form-tab-header-counter {
  padding-right: 0.9em !important; }

.users-edit-form-tab-header {
  position: relative; }

.users-edit-form-counter {
  position: absolute;
  right: 7px;
  width: 2em;
  height: 2em;
  box-sizing: initial;
  border-radius: 50%;
  font-size: 0.71rem;
  line-height: 2em;
  text-align: center;
  color: #FFF;
  background: #5A88C3; }

.edit-user-upload-hint {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 100%;
  padding: 3px;
  transform: translate(-50%, -50%);
  font-size: 0.8rem;
  text-align: center;
  color: #1E1E1E;
  pointer-events: none;
  cursor: pointer; }

.edit-user-input-wrapper {
  margin-bottom: 20px; }
  .edit-user-input-wrapper label {
    color: #1E1E1E; }

.edit-user-input-wrapper-right {
  float: right;
  width: 75%; }
  @media (max-width: 640px) {
    .edit-user-input-wrapper-right {
      float: none;
      width: 100%; } }

.edit-user-input-wrapper-clear {
  clear: both; }

.edit-user-label {
  display: inline-block;
  margin-bottom: 10px; }
  .edit-user-label sup {
    color: #D9534F; }

.edit-user-tip {
  margin-top: 5px;
  color: #B3B3AB;
  font-size: 0.8rem; }
