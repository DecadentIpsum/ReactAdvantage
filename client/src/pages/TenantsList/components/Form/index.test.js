import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';

import Form from './index';

const tenant = {
    tenantName: 'Default',
    name:'Default',
    edition:'Premium',
    id: '1',
    creationTime: '1529242113',
};

it('renders correctly', () => {
    const component = shallow(
        <Form tenant={tenant} />
    );

    expect(component).toMatchSnapshot();
});

test('inputs working correctly', () => {
    const component = mount(
        <Form
            tenant={{}}
            onHide={() => {}}
        />
    );

    const name = 'name';
    const input = component.find(`input[name="${name}"]`);

    input.simulate('change', {
        target: {
            name,
            value: 'test',
        },
    });

    expect(component.state().form[name].value).toBe('test');
});

test('form is passing values to parent component on submit', () => {
    const mockValues = {
        name: 'test',
        editions: 'Standard',
        subscription: "1529242",
        active: true,
    };

    const onSubmit = jest.fn();
    const component = mount(
        <Form
            tenant={tenant}
            onHide={() => {}}
            onSubmit={onSubmit}
        />
    );

    const form = component.state().form;
    const button = component.find('.edit-tenant-save-button').first();

    form.name.value = mockValues.name;
    form.editions.value = mockValues.editions;
    form.subscription.value = new Date(1529242000);
    form.active.value = mockValues.active;
    component.setState(form);

    button.simulate('click');

    expect(onSubmit).toHaveBeenCalledWith(mockValues);
});

test('dropdown is working', () => {
    const component = mount(
        <Form
            tenant={tenant}
            onHide={() => {}}
        />
    );

    const mockValue = 'Standard';

    component.instance().onDropdownChange({ value: mockValue });
    expect(component.state().form.editions.value).toBe(mockValue);
});

test('form is cancelling correctly', () => {
    const onHide = jest.fn();
    const component = mount(
        <Form
            tenant={tenant}
            onHide={onHide}
        />
    );

    component.find('.edit-tenant-cancel-button').first().simulate('click');
    expect(onHide).toHaveBeenCalled();
});
