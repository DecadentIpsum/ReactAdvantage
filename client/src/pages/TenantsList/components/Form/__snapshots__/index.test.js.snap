// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    tenant={
        Object {
            "creationTime": "1529242113",
            "edition": "Premium",
            "id": "1",
            "name": "Default",
            "tenantName": "Default",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <form
        className="edit-tenant"
        onSubmit={[Function]}
>
        <div
                className="edit-tenant-input-wrapper"
        >
                <label
                        className="edit-tenant-label"
                        htmlFor="edit-tenant-name"
                >
                        Name 
                        <sup>
                                *
                        </sup>
                </label>
                <Input
                        autoFocus={true}
                        error=""
                        id="edit-tenant-name"
                        name="name"
                        onChange={[Function]}
                        value="Default"
                />
        </div>
        <div
                className="edit-tenant-input-wrapper"
        >
                <label
                        className="edit-tenant-label"
                >
                        Edition
                </label>
                <Dropdown
                        onChange={[Function]}
                        options={
                                Array [
                                        Object {
                                          "label": "Standard",
                                          "value": "Standard",
                                        },
                                        Object {
                                          "label": "Pro",
                                          "value": "Pro",
                                        },
                                      ]
                        }
                        value="Standard"
                />
        </div>
        <div
                className="edit-tenant-input-wrapper"
        >
                <label
                        className="edit-tenant-label"
                >
                        Subscription end date 
                        <sup>
                                *
                        </sup>
                </label>
                <Calendar
                        minDate={1970-01-18T16:48:12.113Z}
                        onChange={[Function]}
                        value={1970-01-18T16:47:22.113Z}
                />
        </div>
        <div
                className="edit-tenant-input-wrapper"
        >
                <Checkbox
                        checked={false}
                        inputId="edit-tenant-active"
                        label="Active"
                        name="active"
                        onChange={[Function]}
                        value={false}
                />
        </div>
</form>,
      "footer": <div>
        <Button
                className="edit-tenant-cancel-button"
                onClick={undefined}
                secondary={true}
        >
                Cancel
        </Button>
        <Button
                className="edit-tenant-save-button"
                onClick={[Function]}
        >
                Save
        </Button>
</div>,
      "header": "Edit tenant: Default",
      "inPortal": false,
      "modal": true,
      "onHide": undefined,
      "visible": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="edit-tenant-input-wrapper"
>
            <label
                        className="edit-tenant-label"
                        htmlFor="edit-tenant-name"
            >
                        Name 
                        <sup>
                                    *
                        </sup>
            </label>
            <Input
                        autoFocus={true}
                        error=""
                        id="edit-tenant-name"
                        name="name"
                        onChange={[Function]}
                        value="Default"
            />
</div>,
          <div
            className="edit-tenant-input-wrapper"
>
            <label
                        className="edit-tenant-label"
            >
                        Edition
            </label>
            <Dropdown
                        onChange={[Function]}
                        options={
                                    Array [
                                                Object {
                                                  "label": "Standard",
                                                  "value": "Standard",
                                                },
                                                Object {
                                                  "label": "Pro",
                                                  "value": "Pro",
                                                },
                                              ]
                        }
                        value="Standard"
            />
</div>,
          <div
            className="edit-tenant-input-wrapper"
>
            <label
                        className="edit-tenant-label"
            >
                        Subscription end date 
                        <sup>
                                    *
                        </sup>
            </label>
            <Calendar
                        minDate={1970-01-18T16:48:12.113Z}
                        onChange={[Function]}
                        value={1970-01-18T16:47:22.113Z}
            />
</div>,
          <div
            className="edit-tenant-input-wrapper"
>
            <Checkbox
                        checked={false}
                        inputId="edit-tenant-active"
                        label="Active"
                        name="active"
                        onChange={[Function]}
                        value={false}
            />
</div>,
        ],
        "className": "edit-tenant",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="edit-tenant-label"
                htmlFor="edit-tenant-name"
>
                Name 
                <sup>
                                *
                </sup>
</label>,
              <Input
                autoFocus={true}
                error=""
                id="edit-tenant-name"
                name="name"
                onChange={[Function]}
                value="Default"
/>,
            ],
            "className": "edit-tenant-input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name ",
                  <sup>
                    *
</sup>,
                ],
                "className": "edit-tenant-label",
                "htmlFor": "edit-tenant-name",
              },
              "ref": null,
              "rendered": Array [
                "Name ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "*",
                  },
                  "ref": null,
                  "rendered": "*",
                  "type": "sup",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "error": "",
                "id": "edit-tenant-name",
                "name": "name",
                "onChange": [Function],
                "value": "Default",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="edit-tenant-label"
>
                Edition
</label>,
              <Dropdown
                onChange={[Function]}
                options={
                                Array [
                                                Object {
                                                  "label": "Standard",
                                                  "value": "Standard",
                                                },
                                                Object {
                                                  "label": "Pro",
                                                  "value": "Pro",
                                                },
                                              ]
                }
                value="Standard"
/>,
            ],
            "className": "edit-tenant-input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edition",
                "className": "edit-tenant-label",
              },
              "ref": null,
              "rendered": "Edition",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "options": Array [
                  Object {
                    "label": "Standard",
                    "value": "Standard",
                  },
                  Object {
                    "label": "Pro",
                    "value": "Pro",
                  },
                ],
                "value": "Standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="edit-tenant-label"
>
                Subscription end date 
                <sup>
                                *
                </sup>
</label>,
              <Calendar
                minDate={1970-01-18T16:48:12.113Z}
                onChange={[Function]}
                value={1970-01-18T16:47:22.113Z}
/>,
            ],
            "className": "edit-tenant-input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Subscription end date ",
                  <sup>
                    *
</sup>,
                ],
                "className": "edit-tenant-label",
              },
              "ref": null,
              "rendered": Array [
                "Subscription end date ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "*",
                  },
                  "ref": null,
                  "rendered": "*",
                  "type": "sup",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "minDate": 1970-01-18T16:48:12.113Z,
                "onChange": [Function],
                "value": 1970-01-18T16:47:22.113Z,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Checkbox
              checked={false}
              inputId="edit-tenant-active"
              label="Active"
              name="active"
              onChange={[Function]}
              value={false}
/>,
            "className": "edit-tenant-input-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "inputId": "edit-tenant-active",
              "label": "Active",
              "name": "active",
              "onChange": [Function],
              "value": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <form
          className="edit-tenant"
          onSubmit={[Function]}
>
          <div
                    className="edit-tenant-input-wrapper"
          >
                    <label
                              className="edit-tenant-label"
                              htmlFor="edit-tenant-name"
                    >
                              Name 
                              <sup>
                                        *
                              </sup>
                    </label>
                    <Input
                              autoFocus={true}
                              error=""
                              id="edit-tenant-name"
                              name="name"
                              onChange={[Function]}
                              value="Default"
                    />
          </div>
          <div
                    className="edit-tenant-input-wrapper"
          >
                    <label
                              className="edit-tenant-label"
                    >
                              Edition
                    </label>
                    <Dropdown
                              onChange={[Function]}
                              options={
                                        Array [
                                                  Object {
                                                    "label": "Standard",
                                                    "value": "Standard",
                                                  },
                                                  Object {
                                                    "label": "Pro",
                                                    "value": "Pro",
                                                  },
                                                ]
                              }
                              value="Standard"
                    />
          </div>
          <div
                    className="edit-tenant-input-wrapper"
          >
                    <label
                              className="edit-tenant-label"
                    >
                              Subscription end date 
                              <sup>
                                        *
                              </sup>
                    </label>
                    <Calendar
                              minDate={1970-01-18T16:48:12.113Z}
                              onChange={[Function]}
                              value={1970-01-18T16:47:22.113Z}
                    />
          </div>
          <div
                    className="edit-tenant-input-wrapper"
          >
                    <Checkbox
                              checked={false}
                              inputId="edit-tenant-active"
                              label="Active"
                              name="active"
                              onChange={[Function]}
                              value={false}
                    />
          </div>
</form>,
        "footer": <div>
          <Button
                    className="edit-tenant-cancel-button"
                    onClick={undefined}
                    secondary={true}
          >
                    Cancel
          </Button>
          <Button
                    className="edit-tenant-save-button"
                    onClick={[Function]}
          >
                    Save
          </Button>
</div>,
        "header": "Edit tenant: Default",
        "inPortal": false,
        "modal": true,
        "onHide": undefined,
        "visible": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="edit-tenant-input-wrapper"
>
              <label
                            className="edit-tenant-label"
                            htmlFor="edit-tenant-name"
              >
                            Name 
                            <sup>
                                          *
                            </sup>
              </label>
              <Input
                            autoFocus={true}
                            error=""
                            id="edit-tenant-name"
                            name="name"
                            onChange={[Function]}
                            value="Default"
              />
</div>,
            <div
              className="edit-tenant-input-wrapper"
>
              <label
                            className="edit-tenant-label"
              >
                            Edition
              </label>
              <Dropdown
                            onChange={[Function]}
                            options={
                                          Array [
                                                        Object {
                                                          "label": "Standard",
                                                          "value": "Standard",
                                                        },
                                                        Object {
                                                          "label": "Pro",
                                                          "value": "Pro",
                                                        },
                                                      ]
                            }
                            value="Standard"
              />
</div>,
            <div
              className="edit-tenant-input-wrapper"
>
              <label
                            className="edit-tenant-label"
              >
                            Subscription end date 
                            <sup>
                                          *
                            </sup>
              </label>
              <Calendar
                            minDate={1970-01-18T16:48:12.113Z}
                            onChange={[Function]}
                            value={1970-01-18T16:47:22.113Z}
              />
</div>,
            <div
              className="edit-tenant-input-wrapper"
>
              <Checkbox
                            checked={false}
                            inputId="edit-tenant-active"
                            label="Active"
                            name="active"
                            onChange={[Function]}
                            value={false}
              />
</div>,
          ],
          "className": "edit-tenant",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="edit-tenant-label"
                  htmlFor="edit-tenant-name"
>
                  Name 
                  <sup>
                                    *
                  </sup>
</label>,
                <Input
                  autoFocus={true}
                  error=""
                  id="edit-tenant-name"
                  name="name"
                  onChange={[Function]}
                  value="Default"
/>,
              ],
              "className": "edit-tenant-input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name ",
                    <sup>
                      *
</sup>,
                  ],
                  "className": "edit-tenant-label",
                  "htmlFor": "edit-tenant-name",
                },
                "ref": null,
                "rendered": Array [
                  "Name ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "*",
                    },
                    "ref": null,
                    "rendered": "*",
                    "type": "sup",
                  },
                ],
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoFocus": true,
                  "error": "",
                  "id": "edit-tenant-name",
                  "name": "name",
                  "onChange": [Function],
                  "value": "Default",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="edit-tenant-label"
>
                  Edition
</label>,
                <Dropdown
                  onChange={[Function]}
                  options={
                                    Array [
                                                      Object {
                                                        "label": "Standard",
                                                        "value": "Standard",
                                                      },
                                                      Object {
                                                        "label": "Pro",
                                                        "value": "Pro",
                                                      },
                                                    ]
                  }
                  value="Standard"
/>,
              ],
              "className": "edit-tenant-input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edition",
                  "className": "edit-tenant-label",
                },
                "ref": null,
                "rendered": "Edition",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onChange": [Function],
                  "options": Array [
                    Object {
                      "label": "Standard",
                      "value": "Standard",
                    },
                    Object {
                      "label": "Pro",
                      "value": "Pro",
                    },
                  ],
                  "value": "Standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="edit-tenant-label"
>
                  Subscription end date 
                  <sup>
                                    *
                  </sup>
</label>,
                <Calendar
                  minDate={1970-01-18T16:48:12.113Z}
                  onChange={[Function]}
                  value={1970-01-18T16:47:22.113Z}
/>,
              ],
              "className": "edit-tenant-input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Subscription end date ",
                    <sup>
                      *
</sup>,
                  ],
                  "className": "edit-tenant-label",
                },
                "ref": null,
                "rendered": Array [
                  "Subscription end date ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "*",
                    },
                    "ref": null,
                    "rendered": "*",
                    "type": "sup",
                  },
                ],
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "minDate": 1970-01-18T16:48:12.113Z,
                  "onChange": [Function],
                  "value": 1970-01-18T16:47:22.113Z,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Checkbox
                checked={false}
                inputId="edit-tenant-active"
                label="Active"
                name="active"
                onChange={[Function]}
                value={false}
/>,
              "className": "edit-tenant-input-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checked": false,
                "inputId": "edit-tenant-active",
                "label": "Active",
                "name": "active",
                "onChange": [Function],
                "value": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
